miden-lib/asm/miden/kernels/tx/prologue.masm

```diff
@@ -556,101 +554,85 @@ end
 #! note hash.
 #!
 #! Stack: [idx]
-#! Advice stack: [SN, SR, IR, VR, NI, NA, A0, ..., An]
+#! Advice stack: [
+#!      SERIAL_NUMBER,
+#!      SCRIPT_ROOT,
+#!      INPUTS_HASH,
+#!      ASSETS_HASH,
+#!      METADATA,
+#!      ARGS,
+#!      assets_count,
+#!      ASSET_0, ..., ASSET_N,
+#!      block_num,
+#!      SUB_HASH,
+#!      NOTE_ROOT,
+#! ]
 #! Output: []
 #!
 #! Where:
 #! - idx is the index of the input note.
-#! - SN is the serial number of input note `idx`.
-#! - SR is the script root of input note `idx`.
-#! - IR is the inputs root of input note `idx`.
-#! - VR is the vault root of input note `idx`.
-#! - NI is the number of inputs in input note `idx`.
-#! - NA is the number of assets in input note `idx`.
-#! - A0..An are the assets of input note `idx`.
+#! - SERIAL_NUMBER, note's serial.
+#! - SCRIPT_ROOT, note's script root.
+#! - INPUTS_HASH, sequential hash of the padded note's inputs.
+#! - ASSETS_HASH, sequential hash of the padded note's assets.
+#! - METADATA, note's metadata.
+#! - ARGS, user arguments passed to the note.
+#! - inputs_count, note's inputs count.
+#! - assets_count, note's assets count.
+#! - ASSET_0, ..., ASSET_N, padded note's assets.
+#! - block_num, note's creation block number.
+#! - SUB_HASH, the block's sub_hash for which the note was created.
+#! - NOTE_ROOT, the merkle root of the note's tree.
 proc.process_input_note
-    # read core note data
+    # nullifier
     # ---------------------------------------------------------------------------------------------
 
-    # dup the note index
-    dup
-    # => [idx, idx]
-
-    # compute address to store note hash
-    exec.memory::get_consumed_note_ptr
+    dup exec.memory::get_consumed_note_ptr
     # => [note_ptr, idx]
 
-    # compute address to store core note data
     dup exec.memory::get_consumed_note_core_ptr
     # => [note_data_ptr, note_ptr, idx]
 
-    # read nullifier data from the advice provider (serial_num, script_hash, input_hash, asset_hash)
-    padw padw padw adv_pipe hperm adv_pipe hperm
-    # => [PERM, PERM, PERM, note_data_ptr + 4, note_ptr, idx]
+    # read nullifier (SERIAL_NUMBER, SCRIPT_ROOT, INPUTS_HASH, ASSETS_HASH)
+    padw padw padw
+    adv_pipe hperm
+    adv_pipe hperm
```

Comment:
Related to the previous comment: I think it is not exactly correct to say that we are reading "nullifier" here. What we are doing is reading note details and at the same time computing note's nullifier (the way I think about the nullifier is that it is a commitment to a note - in this way, it is similar to note ID).

Reply:
changed to note details